packages/datetime/src/dateRangeInput.tsx

```diff
@@ -463,6 +469,16 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
 
     // Key down
 
+    private handleStartInputKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        this.handleInputKeyDown(e);
+        Utils.safeInvoke(this.props.startInputProps.onKeyDown, e);
+    }
+
+    private handleEndInputKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        this.handleInputKeyDown(e);
+        Utils.safeInvoke(this.props.endInputProps.onKeyDown, e);
+    }
```

Comment:
try:

```ts
onClick={this.handleStartInputEvent}
onKeyDown={this.handleStartInputEvent}
onKeyUp={this.handleStartInputEvent}

// ...

private handleStartInputEvent = (e: React.SyntheticEvent<HTMLInputElement>) => {
    switch (e.type) {
        case "click":   this.handleInputClick(e); break;
        case "keydown": this.handleInputKeyDown(e); break;
        case "keyup":   this.handleInputKeyUp(e); break;
    }

    // get pascalize somewhere, to turn "keydown" -> "KeyDown", "click" -> "Click", etc.
    // or just store those mappings as a constant somewhere, simple enough
    Utils.safeInvoke(this.props.startInputProps[`on${pascalize(e.type)}`], e);
}
```

avoids creating lots of methods. avoids creating new closures on `render`.